conda/common/logic.py

```diff
@@ -262,6 +263,10 @@ def process_solution(self, sat_solution):
         return solution
 
 
+_BIG_NUMBER = maxsize
```

Comment:
We could also use something else here, e.g., `2**31 - 1`.
It just has to be a number that will never be assigned to a SAT variable.
(Using a big number here instead of just `1` and then starting the actual variables from `2` onward is helpful to avoid any of the awkward `1 == True` moments in Python.)

Ideally, we wouldn't have the type ambiguity here, meaning having a `Literal` class whose instances can **either** be a variable number or one of the special values true/false. But that would create to much overhead with CPython.

Still, it is an improvement to have `TRUE`/`FALSE` values explicit rather than `True`/`False` as the latter can lead to ambiguities and hurts comprehension of the code.

Reply:
I can see issues like you describe in a later comment, but I agree with your reasoning here.

Reply:
Would an enum buy any clarity here?  I'm not sure, but it seems like it might fit here.

Reply:
I didn't use an enum due to performance concerns. However thinking about it now, we could do that here: Since we pipe everything that might be a literal through `Clauses._convert`, we could convert a user-facing `common.logic.TRUE` enum to a implementation-detail `common._logic.TRUE`.

Reply:
We'd have to do the inverse conversion, i.e., `common._logic.TRUE` -> `common.logic.TRUE`, for the return values, too, of course.
I think I'll add it as a TODO for now, to revisit it later.

Reply:
Added TODO in https://github.com/conda/conda/pull/9427/commits/ff180eeb0d9b42cac5240c9eb824dc7b01de98a9#diff-13e9ff0832c5112fa7e3819fd7872088R38.